core/src/main/java/org/elasticsearch/search/aggregations/ParsedAggregation.java

```diff
@@ -46,6 +46,10 @@ public final String getName() {
         return name;
     }
 
+    protected void setName(String name) {
+        this.name = name;
```

Comment:
I think that with this protected setter, you can now make the instance member private?